lib/cocoapods/downloader/cache.rb

```diff
@@ -156,7 +170,14 @@ def uncached_pod(request)
 
           podspecs.each do |name, spec|
             destination = path_for_pod(request, :name => name, :params => result.checkout_options)
-            copy_and_clean(target, destination, spec)
+
+            # add file lock
+            destination.parent.mkpath
+            lock_file_path = lock_path_for_pod(request, :name => name, :params => result.checkout_options)
```

Comment:
@yanzhiwei147 shouldn't we unlock the file once we are done?

Something akin to:

```ruby
	    success = file.flock(mode)
	    if success
	      begin
	        yield file
	      ensure
	        file.flock(File::LOCK_UN)
	    end
```

Taken from https://www.safaribooksonline.com/library/view/ruby-cookbook/0596523696/ch06s13.html